tests/generic_config_updater/field_operation_validator_test.py

```diff
@@ -188,6 +188,92 @@ def test_validate_field_operation_illegal__rm_loopback0(self):
         config_wrapper = gu_common.ConfigWrapper()
         self.assertRaises(gu_common.IllegalPatchOperationError, config_wrapper.validate_field_operation, old_config, target_config)
 
+    def test_validate_field_operation_illegal__dataacl_table_type_and_rule(self):
+        old_config = {
+            "ACL_TABLE": {
+                "OPENAI_INGRESS": {
+                    "type": "L2"
+                }
+            },
+            "ACL_RULE": {
+                "OPENAI_INGRESS|RULE_1": {
+                    "ETHER_TYPE": "2048"
+                }
+            }
+        }
+        target_config = {
+            "ACL_TABLE": {
+                "OPENAI_INGRESS": {
+                    "type": "L3V6"
+                }
+            },
+            "ACL_RULE": {
+                "OPENAI_INGRESS|RULE_1": {
+                    "IP_TYPE": "IPV6ANY"
+                }
+            }
+        }
+        config_wrapper = gu_common.ConfigWrapper()
+        self.assertRaises(gu_common.IllegalPatchOperationError,
+                          config_wrapper.validate_field_operation, old_config, target_config)
+
+    def test_validate_field_operation_legal__dataacl_table_type_update(self):
+        old_config = {
+            "ACL_TABLE": {
+                "OPENAI_INGRESS": {
+                    "type": "L3"
+                }
+            },
+            "ACL_RULE": {
+                "OPENAI_INGRESS|RULE_1": {
+                    "IP_TYPE": "IPV6ANY"
+                }
+            }
+        }
+        target_config = {
+            "ACL_TABLE": {
+                "OPENAI_INGRESS": {
+                    "type": "L3V6"
+                }
+            },
+            "ACL_RULE": {
+                "OPENAI_INGRESS|RULE_1": {
+                    "IP_TYPE": "IPV6ANY"
+                }
+            }
+        }
+        config_wrapper = gu_common.ConfigWrapper()
+        config_wrapper.validate_field_operation(old_config, target_config)
+
+    def test_validate_field_operation_legal__dataacl_rule_update(self):
```

Comment:
The three `legal__dataacl` (additional underscore) are typos here?

Reply:
This is the descriptive convention in this unit test file.

Reply:
Thanks!